staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -468,7 +477,17 @@ func (r *crdHandler) createCustomResourceDefinition(obj interface{}) {
 	r.customStorageLock.Lock()
 	defer r.customStorageLock.Unlock()
 	// this could happen if the create event is merged from create-update events
-	r.removeStorage_locked(crd.UID)
+	tearDownFinished := r.removeStorage_locked(crd.UID)
+	if utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) &&
+		utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) {
+		go func() {
```

Comment:
Why is this running in a goroutine? Once the customStorageLock is unlocked, CR requests can go through. Then a CR encoded in a version different than the version recorded in the storageVersion API will be written to the etcd.

Reply:
This code is doing client calls. We have the lock in this moment for the whole CRD apiserver storage map. I.e. a blocked client call might block other requests of other CRDs. This go routine will decouple those.